### DISCUSSION TITLE
Is there a way to avoid calling convertToDatabaseValue during hydration of a custom type?

### DISCUSSION THREAD

**Discussion by alex996** [mikro-orm/mikro-orm#4120] [General]
I figured that `convertToDatabaseValue` on a custom type should only be called when the value is inserted or updated. However, it also gets called when selecting (and in my case, when populating), why?

Is there a way to turn this behavior off? Specifically, I'm referring to [this line](https://github.com/mikro-orm/mikro-orm/blob/075be615664fc05973de2942f942b20747eac0fd/packages/core/src/hydration/ObjectHydrator.ts#L99) in `ObjectHydrator.ts`. I'd like it turned off for all `SELECT` queries involving my custom type. Two reasons: (1) the subproject where I'm re-using this custom type will only read from the DB, but never write; (2) the write requires an environment variable which I cannot make available in my subproject.

In principle, my type is like this:

```ts
class MyType extends Type<string | undefined> {
  convertToDatabaseValue(value: string | undefined) {
    return value ? doSomeProcessingWithSecretEnvVar(value) : value; // this shouldn't get called during hydration!
  }
  convertToJSValue(value: string | undefined) {
    return value ? unwrapTheValueWithNoEnvVarNeeded(value) : value; // this would get called during hydration
  }
}
```

**Comment by B4nan** [comment]
Because the value from the database often comes in a different shape (e.g. arrays, JSON, even Point type or bigint), and we need to make it comparable, otherwise extra updates occur. When I comment that line out, 10 tests are failing because of this.

I guess we can make this configurable via a method on the type class.

**Reply by B4nan** [reply]
Here you go

https://github.com/mikro-orm/mikro-orm/commit/d1ce2408306a9bdd805077ef1a3f9c561eefe982

**Reply by alex996** [reply]
Very quick turnout, thank you! I noticed there is one other call to `convertToDatabaseValue` on [line 168](https://github.com/mikro-orm/mikro-orm/blob/d1ce2408306a9bdd805077ef1a3f9c561eefe982/packages/core/src/hydration/ObjectHydrator.ts#L168) - not sure if it needs to be wrapped with an `if`, but just an FYI.
